### PR TITLE
mimir: Coordinate added in Street

### PR DESCRIPTION
**Pull request by kadhikari**

### CODE REVIEW

File: libs/bragi/src/model.rs.in [hunk 1]
```diff
@@ -85,7 +85,7 @@ impl ToGeom for mimir::Place {
     fn to_geom(&self) -> Option<geojson::Geometry> {
         match self {
             &mimir::Place::Admin(ref admin) => admin.coord.as_ref().map(|c| c.0.to_geom().unwrap()),
```
Comment (TeXitoi) [inline libs/bragi/src/model.rs.in]: This should be
```rust
admin.coord.as_ref().and_then(|c| c.0.to_geom()),

File: src/bin/bano2mimir.rs [hunk 2]
```diff
@@ -76,6 +76,11 @@ impl Bano {
             label: street_name.to_string(),
             administrative_regions: admin.map_or(vec![], |a| vec!(a.clone())),
             weight: 1,
+            coord: mimir::Coord {
+                lat: self.lat,
+                lon: self.lon,
+            }
+
```
Comment (TeXitoi) [inline src/bin/bano2mimir.rs]: useless empty line

**Comment by TeXitoi** [comment]
There is a TODO in model.rs.in:77

**Comment by antoine-de** [comment]
can you add some tests ?

**Comment by kadhikari** [comment]
@TeXitoi I pushed with a fix  of rebase on zip_codes

File: libs/bragi/src/model.rs.in [hunk 3]
```diff
@@ -78,28 +77,28 @@ pub struct GeocodingResponse {
 // and when street will have Coords, we'll have to change Option<Geometry> to
 // Geometry
 trait ToGeom {
-    fn to_geom(&self) -> Option<geojson::Geometry>;
+    fn to_geom(&self) -> geojson::Geometry;
```
Comment (TeXitoi) [inline libs/bragi/src/model.rs.in]: You forgot to remove the TODO

File: libs/mimir/src/objects.rs.in [hunk 4]
```diff
@@ -285,6 +286,17 @@ pub struct AliasParameter{
...
+	pub fn new(lat: f64, lon: f64) -> CoordWrapper {
+		CoordWrapper(geo::Coordinate {x: lon, y: lat})
+	}
+}
+impl ::std::ops::Deref for CoordWrapper {
+	type Target = geo::Coordinate;
+	fn deref(&self) -> &Self::Target {
+		&self.0
+	}
+}
```
Comment (TeXitoi) [inline libs/mimir/src/objects.rs.in]: Damn, that's table indented :( my falt

File: src/admin_geofinder.rs [hunk 5]
```diff
@@ -131,7 +131,7 @@ mod tests {
             label: format!("city {}", offset),
             zip_codes: vec!["421337".to_string()],
             weight: ::std::cell::Cell::new(1),
-            coord: Some(::mimir::CoordWrapper(::geo::Coordinate { x: 4.0 + offset, y: 4.0 + offset })),
+            coord: ::mimir::CoordWrapper(::geo::Coordinate { x: 4.0 + offset, y: 4.0 + offset }),
```
Comment (TeXitoi) [inline src/admin_geofinder.rs]: You can use `::mimir::CoordWrapper::new()` here

File: tests/rubber_test.rs [hunk 6]
```diff
@@ -164,6 +168,10 @@ pub fn rubber_custom_id(mut es: ::ElasticSearchWrapper) {
         assert_eq!(es_elt.find("_id"), es_source.find("id"));
         assert_eq!(es_elt.find("_id"), Some(&to_value("admin:bob")));
         assert_eq!(es_source.find("insee"), Some(&to_value("insee:dummy")));
+        
+        let es_coord = es_source.find("coord").unwrap();
+        assert_eq!(es_coord.find("lat"), Some(&Value::U64(0)));
+        assert_eq!(es_coord.find("lon"), Some(&Value::U64(0)));
```
Comment (TeXitoi) [inline tests/rubber_test.rs]: Better if we put real coords to check something meaningful with lon != lat.

**Comment by TeXitoi** [comment]
OK for me.

**Pull request merged**